api/src/main/java/marquez/db/OpenLineageDao.java

```diff
@@ -104,37 +113,155 @@ default UpdateLineageRow updateBaseMarquezModel(LineageEvent event, ObjectMapper
+                    parentRunRow
+                        .flatMap(run -> jobDao.findJobByUuidAsRow(run.getJobUuid()))
+                        .orElseGet(
+                            () -> {
+                              JobRow newParentJobRow =
+                                  jobDao.upsertJob(
+                                      UUID.randomUUID(),
+                                      getJobType(event.getJob()),
+                                      now,
+                                      namespace.getUuid(),
+                                      namespace.getName(),
+                                      Utils.parseParentJobName(facet.getJob().getName()),
+                                      null,
+                                      jobContext.getUuid(),
+                                      location,
+                                      null,
+                                      inputs);
+                              log.info("Created new parent job record {}", newParentJobRow);
+                              return newParentJobRow;
+                            });
+                log.debug("Found parent job record {}", parentJobRow);
+                if (parentRunRow.isEmpty()) {
+                  RunArgsRow argsRow =
+                      runArgsDao.upsertRunArgs(
+                          UUID.randomUUID(), now, "{}", Utils.checksumFor(ImmutableMap.of()));
+                  ExtendedRunRow newRow =
+                      runDao.upsert(
+                          uuid,
+                          null,
+                          facet.getRun().getRunId(),
+                          now,
+                          parentJobRow.getUuid(),
+                          null,
+                          argsRow.getUuid(),
+                          nominalStartTime,
+                          nominalEndTime,
+                          Optional.ofNullable(event.getEventType())
+                              .map(this::getRunState)
+                              .orElse(null),
+                          now,
+                          namespace.getName(),
+                          parentJobRow.getName(),
+                          parentJobRow.getLocation(),
+                          parentJobRow.getJobContextUuid().orElse(null));
+                  log.info("Created new parent run record {}", newRow);
+                }
+                return parentJobRow;
+              } catch (Exception e) {
+                throw new RuntimeException("Unable to insert parent run", e);
+              }
+            });
+
+    // construct the simple name of the job by removing the parent prefix plus the dot '.' separator
+    String jobName =
+        parentJob
+            .map(
+                p -> {
+                  if (event.getJob().getName().startsWith(p.getName() + '.')) {
+                    return event.getJob().getName().substring(p.getName().length() + 1);
+                  } else {
```

Comment:
_Minor_: We may want to move this in a `DbUtils` class to handle parsing the simple name:

```java
DbUtils.simpleJobNameFor()
```